app/models/movie.rb

```diff
@@ -1,10 +1,16 @@
 class Movie < ApplicationRecord
+  has_many :watch_progresses, dependent: :destroy
+  has_many :user, through: :watch_progresses
```

Comment:
has_many 指定する場合は複数形にしてください。

```suggestion
  has_many :users, through: :watch_progresses
```